pkg/fake_gcloud/lib/mem_datastore.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:gcloud/common.dart';
+import 'package:gcloud/datastore.dart';
+
+/// Implementation of [DatastoreDB] interface with in-memory storage.
+///
+/// Entities are cloned, and should be safe to modify them once they are stored
+/// or read.
+class MemDatastore implements Datastore {
+  final _entities = <Key, Entity>{};
+  int _unusedId = 0;
+
+  @override
+  Future<List<Key>> allocateIds(List<Key> keys) async {
+    return keys.map((k) {
+      if (k.elements.last.id == null) {
+        final elements = List<KeyElement>.from(k.elements);
+        final last = elements.removeLast();
+        elements.add(KeyElement(last.kind, _unusedId++));
+        return Key(elements, partition: k.partition);
+      } else {
+        return k;
+      }
+    }).toList();
+  }
+
+  @override
+  Future<Transaction> beginTransaction({bool crossEntityGroup = false}) async {
+    return _Transaction();
+  }
+
+  @override
+  Future<CommitResult> commit(
+      {List<Entity> inserts,
+      List<Entity> autoIdInserts,
+      List<Key> deletes,
+      Transaction transaction}) async {
+    deletes?.forEach((key) => _entities.remove(key));
+    inserts?.forEach((e) {
+      _entities[e.key] = e;
+    });
+    if (autoIdInserts != null && autoIdInserts.isNotEmpty) {
+      throw UnimplementedError();
```

Comment:
Give it a brief message saying what you haven't implemented :) it helps a lot when we end up debugging this error later :)